### PR TITLE
Properly check 'Finish 5 roller coasters' criteria, fixes #597

### LINKED ISSUE #597: Finish 5 roller coasters objective
The original game allowed the objective to be set on a scenario if there is no money.
There however the following problems:
- [ ] Scenario editor does not check if there are five rollercoasters (you would need to convert a saved game anyway)
- [x] The game didn't exclude null rides from its ride loop check.
- [ ] The game didn't check if the ride was a rollercoaster.
- [ ] The game didn't check if the ride was one of the five partially pre-built rollercoasters.

### PR DESCRIPTION
**Pull request by Gymnasiast**
This fixes the following things:
- The objective was not available when using money (despite it being used on such scenarios in RCT1)
- The scenario editor set the 'partially finished flag' on every single ride
- The scenario editor did not check if there were five finished roller coasters
- When checking the goal, there was no check if the ride was a roller coaster
- When checking the goal, there was no check if the ride was one of the roller coasters to be completed

### CODE REVIEW

File: src/openrct2/scenario/scenario.c [hunk 1]
```diff
@@ -660,38 +660,78 @@ uint32 scenario_rand_max(uint32 max)
...
+    do
+    {
+        if (map_element_get_type(it.element) == MAP_ELEMENT_TYPE_TRACK)
+        {
+            markTrackAsIndestructible = false;
+
             if (isFiveCoasterObjective)
-                it.element->flags |= 0x40;
+            {
+                const rct_ride * ride = get_ride(it.element->properties.track.ride_index);
```
Comment (rwjuk) [inline src/openrct2/scenario/scenario.c]: You've already declared a variable `ride` above - you'll have to either forgo the declaration here or change the variable name.

Comment (Gymnasiast) [inline src/openrct2/scenario/scenario.c]: Well spotted!

File: src/openrct2/scenario/scenario.c [hunk 2]
```diff
@@ -660,38 +660,78 @@ uint32 scenario_rand_max(uint32 max)
...
+static bool scenario_prepare_rides_for_save()
 {
     sint32 isFiveCoasterObjective = gScenarioObjectiveType == OBJECTIVE_FINISH_5_ROLLERCOASTERS;
+    sint32 i;
+    rct_ride * ride;
+    uint8 rcs = 0;
+
+    FOR_ALL_RIDES(i, ride)
+    {
+        const rct_ride_entry * rideEntry = get_ride_entry(ride->subtype);
```
Comment (IntelOrca) [inline src/openrct2/scenario/scenario.c]: null check

**Comment by IntelOrca** [comment]
Not tested it, but it looks good.

**Pull request merged**